### PR TITLE
Strict mode and number to string control

### PR DESCRIPTION
**Pull request by EsTharian**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
-'use strict';
+// ES6 is always in strict mode
```
Comment (sindresorhus) [inline index.js]: This is not correct. It's always strict inside arrow functions, class methods, and modules. The top-level in Node.js is not yet a ES2015 module, so it's not strict by default.

File: index.js [hunk 2]
```diff
@@ -29,7 +29,10 @@ const removeMootSeparators = (string, separator) => {
 };
 
 module.exports = (string, options) => {
-	if (typeof string !== 'string') {
+	if (typeof string === "number") {
+		string = string.toString(); // my filename may be a int or float number
```
Comment (sindresorhus) [inline index.js]: I'm not interested in silently coercing unrelated types. This method accepts a string, if you want it to magically support other things you can do the coercion yourself when calling this method.

**Comment by sindresorhus** [comment]
I appreciate the PR, but it's not something I'm interested in.

File: index.js (hunk 1 above)
Comment (EsTharian) [reply index.js]: You are already using arrow functions and an exported module in this project.

**Pull request closed without merging**